flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -781,6 +787,32 @@ class TaskManager(
     }
   }
 
+  private def handleRequestTaskManagerLog
+  (sender: ActorRef, requestType: LogTypeRequest, jobManager: ActorRef):
+  Unit = {
+    if (blobService.isDefined && jobManager.path.address.host.isDefined) {
```

Comment:
this check is made specifically for the start-local.sh script, for which the jobManager does not have a defined address.